src/Nri/Ui/Checkbox/V6.elm

```diff
@@ -0,0 +1,640 @@
+module Nri.Ui.Checkbox.V6 exposing
+    ( checkbox, disabled, locked
+    , Selected(..), binary
+    )
+
+{-|
+
+
+# Changes from V5:
+
+  - Complete redo of the api for more precise data modeling.
+  - Adds an 0.4 opacity to the label of disabled checkbox.
+
+@docs checkbox, disabled, locked
+
+@docs Selected, binary
+
+-}
+
+import Accessibility.Styled as Html
+import Accessibility.Styled.Aria as Aria
+import Accessibility.Styled.Style
+import Accessibility.Styled.Widget as Widget
+import Css exposing (..)
+import Css.Global
+import Html.Events
+import Html.Styled
+import Html.Styled.Attributes as Attributes exposing (css)
+import Html.Styled.Events as Events
+import Json.Decode
+import Nri.Ui.Colors.V1 as Colors
+import Nri.Ui.Fonts.V1 as Fonts
+import Nri.Ui.Html.V3 as HtmlExtra exposing (defaultOptions)
+import Svg
+import Svg.Attributes
+
+
+{-| Create a regular checkbox:
+
+    - `identifier` is a unique id for this checkbox.
+    - `label` is the label of the checkbox.
+    - `anonymous` is whether the label is shown or not.
+    - `selected` is the state of the checkbox.
+    - `onChange` is the handler called when the state changes.
+
+-}
+checkbox :
+    { identifier : String
+    , label : String
+    , anonymous : Bool
+    , selected : Selected
+    , onChange : Bool -> msg
+    }
```

Comment:
Should we switch to the new style of API of having `checkbox : List Checkbox.Attribute -> Html` ?   I guess label, selected, onChange should always be required,   but I think identifier could be option (generate from the label if not given),  anonymous (hidden label) is only needed occasionally,  and disabled/locked could make sense as an attribute instead of as a separate function.   Switching to attributes would also make it nicer for selected, since we could have both `selected : Bool -> Attribute` and `maybeSelected : Selected -> Attribute`

Reply:
I have a few concerns with that approach:
- You loose the ability to prevent faulty configurations.
- If one makes a faulty configuration, it won't necessarily be obvious what the result is.
- You can set the same attribute twice, overriding the first without knowing.
- More lookups in documentation to see what attribute to use.

Personally, I prefer to just be explicit about the API, but if NRI already took a stand on the issue, I'm happy to oblige.